### PR TITLE
Resolve image version from "latest" in prepare_environment

### PR DESCRIPTION
**Pull request by dsrivastavv**
Fix issue where TestResult message has "latest" in image name rather than the actual version

### CODE REVIEW

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -363,6 +363,11 @@ def _prepare_environment(  # noqa: C901
                     else:
                         existing_location = node_runbook.location
 
+                # resolve Latest to specified version
+                node_runbook.marketplace = self._parse_marketplace_image(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: It should be called, only if the marketplace is not empty.

Comment (dsrivastavv) [reply lisa/sut_orchestrator/azure/platform_.py]: yes. Just saw the test failures on this. Both location and marketplace need to be specified

File: lisa/sut_orchestrator/azure/platform_.py [hunk 2]
```diff
@@ -363,6 +363,12 @@ def _prepare_environment(  # noqa: C901
                     else:
                         existing_location = node_runbook.location
 
+                    if node_runbook.marketplace:
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: I think it again, it should be put before the last return, and in the `if environment.runbook.nodes_requirement:`. There are some different scenarios. That's the right place to resolve the image name. You can add a new for loop there.

Comment (dsrivastavv) [reply lisa/sut_orchestrator/azure/platform_.py]: Do you mean just before `return is_success` on line 482?

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: yes

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**